src/TripSegment/index.js

```diff
@@ -0,0 +1,313 @@
+  overflow: hidden;
+  flex-wrap: wrap;
+  padding: ${({ theme, shown }) => shown && `${theme.orbit.spaceMedium} 0`};
+  max-height: ${({ shown, contentHeight }) => (shown ? `${contentHeight}px` : "0")};
+  animation: ${({ shown, theme, contentHeight }) =>
+    shown
+      ? `${theme.orbit.durationFast}${ToggleUp(contentHeight)} linear;
+        `
+      : `${theme.orbit.durationFast}${ToggleDown(contentHeight)} linear;
+        `};
+  transition: max-height ${({ theme }) => theme.orbit.durationFast} linear,
+    padding ${({ theme }) => theme.orbit.durationFast} linear;
+
+  font-size: ${({ theme }) => theme.orbit.spaceSmall};
+  ${({ shown }) =>
+    shown &&
+    css`
+      padding-top: ${({ theme }) => theme.orbit.spaceSmall};
+      &:before {
+        content: "";
+        width: 100%;
+        position: absolute;
+        top: 0;
+        height: 1px;
+        background: ${({ theme }) => theme.orbit.paletteCloudNormal};
+      }
+    `};
+  ${Item} {
+    margin-bottom: ${({ theme }) => theme.orbit.spaceXXSmall};
+  }
+`;
+
+ChildrenWrapper.defaultProps = {
+  theme: defaultTokens,
+};
+
+const Icon = ({ type }) => {
+  switch (type) {
+    case "airline":
+      return <Airplane size="small" color="secondary" />;
+    case "train":
+      return <Train size="small" color="secondary" />;
+    case "bus":
+      return <Bus size="small" color="secondary" />;
+    default:
+      return <Airplane size="small" color="secondary" />;
+  }
+};
+
+class TripSegment extends React.PureComponent<Props, State> {
+  state = {
+    shown: false,
+  };
+
+  componentDidMount() {
+    setTimeout(() => {
+      this.setHeight();
+    }, 250);
+  }
+
+  setHeight() {
+    const { node } = this;
+    if (node instanceof HTMLElement) {
+      const element = Array.from(node.childNodes).find((item, index) => index === 0 && item);
+      this.contentHeight = Object.values(element.childNodes).reduce(
+        // $FlowExpected
+        (acc, child) => acc + child.offsetHeight,
+        0,
+      );
+    }
+  }
+
+  handleToggle = () => {
+    const { onClick } = this.props;
+    this.setState(prevState => ({
+      shown: !prevState.shown,
+    }));
+
+    if (onClick) {
+      onClick();
+    }
+  };
+
+  contentHeight = 0;
+  node = {};
+
+  render() {
+    const {
+      children,
+      departure,
+      departureTime,
+      arrival,
+      type,
+      arrivalTime,
+      duration,
+      carrier,
+    } = this.props;
+    const { shown } = this.state;
+    const { contentHeight } = this;
+
+    return (
+      <Wrapper>
+        <Milestone>
+          <Icon type={type} />
+          <ArrowDecor />
+        </Milestone>
+        <StyledSegment>
+          <WrapperInner shown={shown} onClick={this.handleToggle}>
+            <Flights>
+              <FlightPart>
+                <Time>
+                  <Text type="attention">{arrivalTime}</Text>
+                </Time>
+                <Text type="primary">
+                  {arrival.city}
```

Comment:
I think that we can set `arrival` prop to accept a string that consists off `city` and `code` in just one prop.